qltui.py

```diff
@@ -0,0 +1,743 @@
+import os
+import ast
+import pickle
+import re
+import six
+import argparse
+from json2html import *
+import pdfkit
+from pyfx import Controller
+import json
+from pprint import pprint
+from pyfx.model import DataSourceType
+from datetime import datetime
+
+import questionary
+from questionary import Validator, ValidationError, prompt
+try:
+    from termcolor import colored
+except ImportError:
+    colored = None
+
+
+from qiling.const import os_map, arch_map, verbose_map
+from qiling.extensions.coverage import utils as cov_utils
+
+motd = """
+    ██████     ███  ████   ███                     
+  ███░░░░███  ░░░  ░░███  ░░░                      
+ ███    ░░███ ████  ░███  ████  ████████    ███████
+░███     ░███░░███  ░███ ░░███ ░░███░░███  ███░░███
+░███   ██░███ ░███  ░███  ░███  ░███ ░███ ░███ ░███
+░░███ ░░████  ░███  ░███  ░███  ░███ ░███ ░███ ░███
+ ░░░██████░██ █████ █████ █████ ████ █████░░███████
+   ░░░░░░ ░░ ░░░░░ ░░░░░ ░░░░░ ░░░░ ░░░░░  ░░░░░███
+                                           ███ ░███
+                                          ░░██████ 
+                                           ░░░░░░
+"""
+
+ERROR_COLOR = "red"
+
+HEADING_COLOR = "green"
+
+OUTPUT_COLOR = "blue"
+
+TITLE_COLOR = "blue"
+
+prog = os.path.basename(__file__)
+
+HEX_REGEX = r'^(0[xX])[a-fA-F0-9]+$'
+
+
+class Callback_Functions():
+    """
+    Callback Functions for Hook Operation
+    """
+
+    @staticmethod
+    def read_mem_2_args(ql, user_data):
```

Comment:
Code duplication for `read_mem_*_args` seems to be avoidable by referring all methods to a single implementation. Anyway all args other than `user_data` are discarded.

A cleaner way to do that would be defining just one method that receives an arbitrary number of arguments, but expects a named `user_data` arg at the end. This way all methods would be consolidated into just one method, and calling them would be done by expending a `**kwargs` rather than just `*args` (at the bottom line of this file)